core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -748,4 +740,20 @@ private Instant getNotAfter(final JsonObject trustedCa) {
         }
         return instant;
     }
+
+    private boolean isValidTenantConfig(final JsonObject trustedCa) {
+
+        final String subjectDn = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class);
+        final String publicKey = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, String.class);
+        final Instant notBefore = getNotBefore(trustedCa);
+        final Instant notAfter = getNotAfter(trustedCa);
+
+        if (Strings.isNullOrEmpty(subjectDn) ||
+                // if the public key is present, then
```

Comment:
is there a case where the public key does not need to be present?

Reply:
no, there is not. do you mean we check for its absence?

Reply:
I would expect this method to throw an IllegalArgumentException if the passed in JSON object did not contain a public key, wouldn't you?

Reply:
sure 👍